src/nvim/window.c

```diff
@@ -4923,7 +4923,7 @@ void scroll_to_fraction(win_T *wp, int prev_height)
   invalidate_botline_win(wp);
 
   if (wp->w_buffer->terminal) {
-    terminal_resize(wp->w_buffer->terminal, 0, wp->w_height);
+    terminal_resize_into(wp->w_buffer->terminal, wp);
```

Comment:
`0` has a special meaning here, so the logic changed now. Same for `win_new_width`.